cmd/age/read_password_windows.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+// +build windows
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/term"
+)
+
+func readPassphraseFromTerminal() ([]byte, error) {
+	conin, err := windows.UTF16PtrFromString("CONIN$")
+	if err != nil {
+		return nil, err
+	}
+	tty, err := windows.CreateFile(conin, windows.GENERIC_READ|windows.GENERIC_WRITE, windows.FILE_SHARE_READ, nil, windows.OPEN_EXISTING, 0, 0)
```

Comment:
I don't think you need x/sys/windows. `os.Open` should work too.

Reply:
I haven't tried it in `age`, but in another program and it didn't work. See https://github.com/golang/go/issues/46164#issuecomment-851978139 for more info.

Reply:
@codesoap os.OpenFile works for me (windows 10, go 1.12) 

See https://play.golang.org/p/FTJwXE0o6AP

Reply:
That's interesting, thanks. This may even be relevant for https://github.com/golang/go/issues/46164. I'll play around with it a bit as well.

Reply:
I can now confirm that using `os.OpenFile("CONIN$", os.O_RDWR, 0777)` works and `os.OpenFile("CONIN$", os.O_RDONLY, 0777)` does not. Cool find! However, this seems quite odd to me, since `CONIN$` is only supposed to read anyway. Is this to be considered a bug of Windows?!

Reply:
That's odd, SetConsoleMode says it only needs GENERIC_READ access. It would be nice to know which syscall in term.ReadPassword was throwing the access denied error.

Reply:
If I use the "long way" of reading a password with x/term, we can see, that the error seems to originate from [MakeRaw's implementation on Windows](https://github.com/golang/term/blob/a79de5458b56c188f4fc267a58014ac25fec956a/term_windows.go#L23). So I guess you are correct and `SetConsoleMode` probably is the culprit. Here is a demo: https://play.golang.org/p/qRpuBKj2poq